Lib/doctest.py

```diff
@@ -1097,6 +1097,14 @@ def _find_lineno(self, obj, source_lines):
         if inspect.iscode(obj):
```

Comment:
Change line 1097 into:
        if isinstance(obj, str) and source_lines is not None:
to fix a small bug